core/src/main/scala/akka/persistence/jdbc/state/scaladsl/DurableStateStore.scala

```diff
@@ -0,0 +1,52 @@
+package akka.persistence.jdbc.state.scaladsl
+
+import scala.concurrent.{ ExecutionContext, Future }
+import slick.jdbc.{ JdbcBackend, JdbcProfile }
+import akka.Done
+import akka.persistence.state.scaladsl.{ DurableStateUpdateStore, GetObjectResult }
+import akka.serialization.Serialization
+import akka.persistence.jdbc.state.{ AkkaSerialization, DurableStateQueries }
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+import akka.persistence.jdbc.state.DurableStateTables
+import akka.dispatch.ExecutionContexts
+
+class DurableStateStore[A](
+    db: JdbcBackend#Database,
+    profile: JdbcProfile,
+    durableStateConfig: DurableStateTableConfiguration,
+    serialization: Serialization)(implicit ec: ExecutionContext)
+    extends DurableStateUpdateStore[A] {
```

Comment:
I need to understand a bit more on what exactly does this class `DurableStateStoreProvider` offer ? We had one in in memory implementation as well IIRC

Reply:
That is the factory of the javadsl/scaladsl DurableStateStore. It's the thing that is configured, see corresponding test in Akka https://github.com/akka/akka/blob/dev-durable-state/akka-persistence-typed-tests/src/test/scala/akka/persistence/typed/state/scaladsl/DurableStateBehaviorReplySpec.scala#L27

It can be very similar to https://github.com/akka/akka-persistence-jdbc/blob/master/core/src/main/scala/akka/persistence/jdbc/query/JdbcReadJournalProvider.scala